lib/pad.js

```diff
@@ -0,0 +1,33 @@
+var pry = require('pryjs');
+
+Array.prototype.pad = function(num, word){
+  var arrLength = this.length;
+  var array = this;
+  var timesDo = checkInput(num, array)
+  var newArray = [];
+  for(i=0; i < timesDo; i++){
+    if(array[i] == undefined){
+      checkWord(word, newArray)
+    } else {
+      newArray.push(array[i])
+    }
+  }
+  return newArray
+}
+
+function checkInput(num, array) {
+  if (num <= array.length){
+    return array.length
+  } else {
+    return num
+  }
+}
+
+function checkWord(word, newArray) {
+  if(word == undefined){
+    newArray.push(null)
+  } else {
+    newArray.push(word)
+  }
+  return newArray
+}
```

Comment:
@CjMoore you're right - there are fewer agreed upon conventions with JS than Ruby, but some people do choose to write object oriented JS. that being said, some people do not. breaking things down into smaller pieces to make things more single responsibility and easier to test is always a good idea in my opinion 👍 it's also much easier to read!